### PR TITLE
Order trips for date

### PR DESCRIPTION
**Pull request by scara82**
This pr:
- Order the trips in index for dates;
- Split the list in 2: old and current trips;
- Add some style.

@elle I'm not sure if I should test it because I can't create trip in the past. Any suggestions? Thanks!

screen shot 2018-02-27 at 9 39 08 pm

 
Trello card:
[https://trello.com/c/giq03um9](url)
[https://trello.com/c/5vbHu4wO](url)

### CODE REVIEW

File: app/assets/stylesheets/base/_forms.scss [hunk 1]
```diff
@@ -36,3 +36,9 @@ span.error {
 #trip_start_date_3i {
   max-width: fit-content;
 }
+
+.trips_table_index {
```
Comment (elle) [inline app/assets/stylesheets/base/_forms.scss]: Does this belong in `_forms.scss`? or would it be more suitable in `_tables.scss`?

File: app/controllers/trips_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class TripsController < ApplicationController
   def index
-    @trips = Trip.all
+    @current_trips = Trip.where("end_date > ?", Time.now).order("start_date")
+    @past_trips = Trip.where("end_date <= ?", Time.now)
```
Comment (elle) [inline app/controllers/trips_controller.rb]: A few suggestions:
1. move the query to the model (use named scope), so that you can call `Trip.current` and `Trip.past`
2. make the order a scope as well, i.e. `Trip.by_start_date`
3. consider creating a `TripsPresenter`, which has current and past trips.

Comment (elle) [reply app/controllers/trips_controller.rb]: Once you have the scopes on the model, you can stub them in a spec. For example:
```
past_trip = build_stubbed(:trip, start_date: 1.month.ago)
allow(Trip).to receive(:past).and_return([past_trip])
```

Comment (elle) [reply app/controllers/trips_controller.rb]: Lastly, never use `Time.now` which uses your machine time. Instead either use `Time.current` or `Time.zone.now`, which will use the application time zone

**Review by elle — COMMENTED** [review]
Re: layout: various websites use tabs to display current/past trips. But really up to you :)

File: app/controllers/trips_controller.rb (hunk 2 above)
Comment (elle) [reply app/controllers/trips_controller.rb]: Actually, thinking about this more, I would leave creating the presenter for now. But will still do points 1 and 2

Comment (scara82) [reply app/controllers/trips_controller.rb]: - Added scope in the model in sha:`a0599c2` & add the scope into the index#TripsConroller in  she: `2bd2df1` and replace `Time.now` with `Time.current`. Not sure the convention for the position of scope, I put it just before the private methods;
- Added the a sub_header with tab for `current trips` , `past trips` and `all trips` has been made in few commits, I will add a pic in the PR description.
- Test for landing in the index page and all the 3 scenarios (last commit).

File: app/assets/stylesheets/base/_forms.scss (hunk 1 above)
Comment (scara82) [reply app/assets/stylesheets/base/_forms.scss]: Done in sha `3906085` with fixup

**Comment by scara82** [comment]
@elle thanks for yours comments and suggestions, I'm merging now. 🚀

File: app/assets/javascripts/trips.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+function showPastTrips(){
+  document.getElementById("past_trips").style.display = "block";
+  document.getElementById("current_trips").style.display = "none";
+  document.getElementById("all_trips").style.display = "none";
+}
+
+function showAllTrips(){
+  document.getElementById("all_trips").style.display = "block";
+  document.getElementById("past_trips").style.display = "none";
+  document.getElementById("current_trips").style.display = "none";
```
Comment (elle) [inline app/assets/javascripts/trips.js:16]: If you are creating a function to hide all buttons on the page, and then just show one of the buttons, then this function can be re-used here.

File: app/models/trip.rb [hunk 4]
```diff
@@ -13,6 +13,10 @@ class Trip < ApplicationRecord
 
   delegate :name, to: :school, prefix: true
 
+  scope :current, -> { where("end_date > ?", Time.current) }
+  scope :past, -> { where("end_date <= ?", Time.current) }
+  scope :by_start_date, -> { order("start_date") }
```
Comment (elle) [inline app/models/trip.rb:18]: Change the scope to use `def self.current` syntax

File: config/initializers/assets.rb [hunk 5]
```diff
@@ -12,3 +12,4 @@
 # application.js, application.css, and all non-JS/CSS in the app/assets
 # folder are already added.
 # Rails.application.config.assets.precompile += %w( admin.js admin.css )
+Rails.application.config.assets.precompile += ["jquery/dist/jquery.js"]
```
Comment (elle) [inline config/initializers/assets.rb:15]: Can we get jQuery to be compiled without adding this line?
Also, you are not currently using jQuery, so why is this needed?
